functorch/functorch/csrc/BatchRulesLinearAlgebra.cpp

```diff
@@ -382,6 +382,39 @@ fourOutputs linalg_lstsq_batch_rule(
   return std::make_tuple(res, 0, res_1, res_1_bdim, res_2, res_2_bdim, res_3, res_3_bdim);
 }
 
+std::tuple<Tensor, c10::optional<int64_t>>
+matrix_rank_atol_rtol_tensor_batch_rule(
+    const Tensor& input, optional<int64_t> input_bdim, const optional<Tensor>& atol, const optional<int64_t> atol_bdim,
+    const optional<Tensor>& rtol, const optional<int64_t> rtol_bdim, bool hermitian) {
+  const auto input_logical_rank = rankWithoutBatchDim(input, input_bdim);
+  TORCH_CHECK(input_logical_rank >= 2,
+            "torch.linalg.matrix_rank: The input tensor input must have at least 2 dimensions.");
+
+  // atol and rtol's dims must match the number of batch dims of input, which is its dim - 2 (for the matrix dimensions)
+  const auto input_logical_num_bdims = input_logical_rank - 2;
+  const int64_t atol_logical_num_bdims = atol.has_value() ? rankWithoutBatchDim(*atol, atol_bdim) : 0;
+  const int64_t rtol_logical_num_bdims = rtol.has_value() ? rankWithoutBatchDim(*rtol, rtol_bdim) : 0;
```

Comment:
Is this saying that atol_logical_num_bdims must be equal to input_logical_num_bdims?

Reply:
Oh nvm I see

Reply:
This is confusing. I added to that comment to be more verbose and added another comment after that to make it a little clearer what's happening